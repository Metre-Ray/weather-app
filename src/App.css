body {
  display: grid;
  align-items: center;
  height: 100vh;
  background-color: black;
  color: white;
}

.app {
  display: grid;
  grid-template-columns: 1fr 1fr;
  width: 90%;
  max-width: 1000px;
  margin: 0 auto;
  text-align: center;
  box-shadow: 0 0 5px rgb(255, 213, 135);
}

.status {
  grid-column: 1 / 3;
  padding: 5px;
}
