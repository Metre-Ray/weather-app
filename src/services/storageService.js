const STORAGE_KEY = 'weather-search-app';

const storageService = {
  setItems: (item1, item2) => {
    localStorage.setItem(STORAGE_KEY, `${item1} ; ${item2}`);
  },
  getItems: () => {
    const tempItem = localStorage.getItem(STORAGE_KEY);
    if (!tempItem) return [null, null];
    const [city, country] = tempItem.split(';').map((el) => el.trim());
    return [city, country];
  },
  clearStorage: () => {
    localStorage.clear();
  }
}

export default storageService;
