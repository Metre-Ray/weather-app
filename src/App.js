import React, { Component } from 'react'
import Form from './components/Form';
import Titles from './components/Titles';
import Weather from './components/Weather';
import storageService from './components/services/storageService';

import './App.css';

const API_KYE = '8bf768353480692e06febb4c621734e9';
// const url = `https://api.openweathermap.org/data/2.5/weather?q=London,uk&APPID=${API_KYE}`;
export default class App extends Component {
  state = {
    tempreture: undefined,
    city: undefined,
    country: undefined,
    humidity: undefined,
    pressure: undefined,
    windSpeed: undefined,
    description: undefined,
    longitude: undefined,
    latitude: undefined,
    base: undefined,
    error: undefined
  }

  getWeather = async (event) => {
    event.preventDefault();
    const city = event.target.elements.city.value;
    const country = event.target.elements.country.value;
    if (!city || !country) {
      this.setState({
        tempreture: undefined,
        city: undefined,
        country: undefined,
        humidity: undefined,
        pressure: undefined,
        windSpeed: undefined,
        description: undefined,
        longitude: undefined,
        latitude: undefined,
        base: undefined,
        error: 'Please, enter city and country'
      })
      return;
    }
    const url = `https://api.openweathermap.org/data/2.5/weather?q=${city},${country}&APPID=${API_KYE}`;
    const api_call = await fetch(url);
    const data = await api_call.json();
    if (data.cod !== 200) {
      this.setState({
        tempreture: undefined,
        city: undefined,
        country: undefined,
        humidity: undefined,
        pressure: undefined,
        windSpeed: undefined,
        description: undefined,
        longitude: undefined,
        latitude: undefined,
        base: undefined,
        error: `Sorry, ${data.message}`
      })
      return;
    }
    this.setState({
      temperature: data.main.temp,
      pressure: data.main.pressure,
      humidity: data.main.humidity,
      windSpeed: data.wind.speed,
      longitude: data.coord.lon,
      latitude: data.coord.lat,
      city: data.name,
      country: data.sys.country,
      description: data.weather[0].description,
      base: data.base,
      error: ''
    });
    storageService.setItems(city, country);
  }

  componentDidMount() {
    const [city, country] = storageService.getItems();
    this.value1 = city;
    this.value2 = country;
    console.log('mount', this.value1);
  }

  render() {
    return (
      <div className="app">
        <Titles></Titles>
        <Form getWeather={this.getWeather} ></Form>
        <Weather {...this.state}></Weather>
      </div>
    )
  }
}

// Задачи:
// 1. стилизовать
// 2. решить, куда поместить API ключ (может в базу firebase?? - но для доступа к ней
// тоже нужен ключ; тогда на хероку?) - почитать
