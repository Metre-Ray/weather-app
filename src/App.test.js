import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow, render, mount } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import storageService from './services/storageService';

Enzyme.configure({ adapter: new Adapter() });

it('app renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("match shallow snapshot", () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});

it("match render snapshot", () => {
  const wrapper = render(<App />);
  expect(wrapper).toMatchSnapshot();
});

it("match mount snapshot", () => {
  const wrapper = mount(<App />);
  expect(wrapper).toMatchSnapshot();
});

it("setState should be called in componentDidMount one time", () => {
  const inst = shallow(<App />).instance();
  jest.spyOn(inst, 'setState').mockImplementation();
  inst.componentDidMount();
  expect(inst.setState).toHaveBeenCalled();
  expect(inst.setState).toHaveBeenCalledTimes(1);
  //inst.setState.mock.calls[0][0]
});

it("mock storage service", () => {
  const inst = shallow(<App />).instance();
  // TODO
});
