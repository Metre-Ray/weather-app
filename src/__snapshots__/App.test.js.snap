// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match mount snapshot 1`] = `
<App>
  <div
    className="app"
  >
    <Titles>
      <div
        className="titlesContainer"
      >
        <h1>
          Weather search
        </h1>
        <p>
          Find out weather, temperature and other conditions
        </p>
      </div>
    </Titles>
    <Form
      getWeather={[Function]}
      value1={null}
      value2={null}
    >
      <form
        action=""
        className="formContainer"
        onSubmit={[Function]}
      >
        <input
          className="input"
          defaultValue={null}
          name="city"
          placeholder="City..."
          type="text"
        />
        <input
          className="input"
          defaultValue={null}
          name="country"
          placeholder="Country..."
          type="text"
        />
        <button
          className="button"
        >
          Get Weather
        </button>
      </form>
    </Form>
    <Weather
      defaultCity={null}
      defaultCountry={null}
    >
      <div
        className="weather"
      />
    </Weather>
  </div>
</App>
`;

exports[`match render snapshot 1`] = `
<div
  class="app"
>
  <div
    class="titlesContainer"
  >
    <h1>
      Weather search
    </h1>
    <p>
      Find out weather, temperature and other conditions
    </p>
  </div>
  <form
    action=""
    class="formContainer"
  >
    <input
      class="input"
      name="city"
      placeholder="City..."
      type="text"
    />
    <input
      class="input"
      name="country"
      placeholder="Country..."
      type="text"
    />
    <button
      class="button"
    >
      Get Weather
    </button>
  </form>
  <div
    class="weather"
  />
</div>
`;

exports[`match shallow snapshot 1`] = `
<div
  className="app"
>
  <Titles />
  <Form
    getWeather={[Function]}
    value1={null}
    value2={null}
  />
  <Weather
    defaultCity={null}
    defaultCountry={null}
  />
</div>
`;
